google/cloud/bigquery/v2/minimal/internal/job_client_test.cc

```diff
@@ -93,6 +94,32 @@ TEST(JobClientTest, GetJobFailure) {
                                HasSubstr("bad-request-error")));
 }
 
+TEST(JobClientTest, ListJobs) {
+  auto mock = std::make_shared<MockBigQueryJobConnection>();
+  EXPECT_CALL(*mock, options);
+
+  EXPECT_CALL(*mock, ListJobs).WillOnce([](ListJobsRequest const& request) {
+    EXPECT_EQ(request.project_id(), "test-project-id");
+    return google::cloud::internal::MakePaginationRange<
+        StreamRange<ListFormatJob>>(
+        ListJobsRequest{},
+        [](ListJobsRequest const&) {
+          return StatusOr<ListJobsResponse>(
+              Status(StatusCode::kPermissionDenied, "denied"));
+        },
+        [](ListJobsResponse const&) { return std::vector<ListFormatJob>{}; });
```

Comment:
There is a simpler way to write this using the `google::cloud::mocks` library.

```cc
#include "google/cloud/mocks/mock_stream_range.h"
#include "google/cloud/internal/make_status.h"
```
```cc
return mocks::MakeStreamRange<ListFormatJob>({}, internal::PermissionDeniedError("denied"));
```

I think you should get the dep on the `mocks` library for free through the `google_cloud_cpp_testing` library, but it might take edits to the build files.